src/docfx/config/ops/OpsConfigAdapter.cs

```diff
@@ -47,18 +78,100 @@ public static JObject Load(FileResolver fileResolver, SourceInfo<string> name, s
                 {
                     ["defaultLocale"] = GetDefaultLocale(docset.site_name),
                 },
+                ["monikerDefinition"] = MonikerDefinitionApi,
+                ["markdownValidationRules"] = $"{MarkdownValidationRulesApi}{metadataServiceQueryParams}",
+                ["metadataSchema"] = new JArray(
+                    Path.Combine(AppContext.BaseDirectory, "data/schemas/OpsMetadata.json"),
+                    $"{MetadataSchemaApi}{metadataServiceQueryParams}"),
             };
         }
 
-        private static string ResolveFile(FileResolver fileResolver, SourceInfo<string> name, string url)
+        public static async Task<HttpResponseMessage> InterceptHttpRequest(ErrorLog errorLog, HttpRequestMessage request)
+        {
+            foreach (var (baseUrl, rule) in s_apis)
+            {
+                if (request.RequestUri.OriginalString.StartsWith(baseUrl))
+                {
+                    return new HttpResponseMessage { Content = new StringContent(await rule(errorLog, request.RequestUri)) };
+                }
+            }
+            return null;
+        }
+
+        private static Task<string> GetMonikerDefinition(ErrorLog errorLog, Uri url)
+        {
+            return Fetch($"{s_buildServiceEndpoint}/v2/monikertrees/allfamiliesproductsmonikers", s_opsHeaders);
+        }
+
+        private static async Task<string> GetMarkdownValidationRules(ErrorLog errorLog, Uri url)
         {
             try
             {
-                return fileResolver.ReadString(new SourceInfo<string>(url));
+                var headers = GetValidationServiceHeaders(url);
+
+                return await Fetch($"{s_validationServiceEndpoint}/rules/content", headers);
             }
-            catch (DocfxException ex) when (ex.InnerException is HttpRequestException hre && hre.Message.Contains("404"))
+            catch (Exception ex)
             {
-                throw Errors.DocsetNotProvisioned(name).ToException();
+                Log.Write(ex);
+                errorLog.Write(Errors.ValidationIncomplete());
+                return "{}";
+            }
+        }
+
+        private static async Task<string> GetMetadataSchema(ErrorLog errorLog, Uri url)
+        {
+            try
+            {
+                var headers = GetValidationServiceHeaders(url);
+                var rules = Fetch($"{s_validationServiceEndpoint}/rules", headers);
```

Comment:
is it possible to have two explicit get URL calls instead of having two implicit calls within one public get URL?

Reply:
That would require changes outside of this adapter layer because docfx core only recognize one JSON file, the adapter feels the right place to bridge the gap between validation service APIs and docfx.